enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,418 @@
+ 
+More and more customers/partners want to enable hardware and or software on OpenShift that need kernel drivers, which are currently not (and may never be) upstream   and out-of-tree. 
+We want to provide a unified way to support out-of-tree kernel drivers (multiple) on OpenShift for day-1 (long term) and day-2 (short term). It has to work for classical (RHEL7, 8, Fedora) and container based operating systems (RHCOS, FCOS) in the same way. 
+Many customers/partners are using dkms/akmods as the solution to build, rebuild modules on kernel changes. Adopting dkms is not a solution; we need to  create an own acceptable solution for RHEL and Fedora. 
+ 
+#### Fill the gap of providing drivers that are not yet upstream or never be
+
+For the DriverContainer we need to cover several stages of driver packaging. 
+
+* **github repository or archive** The driver is only available as  source code and is not packaged in any way, this is where KVC can help
+* **kmod-{vendor}-src.rpm** The next step is a source RPM package that can be recompiled with rpmbuild. This is also the base for akmods, dkms
+* **precompiled-{vendor}-{kernelversion}.rpm** Precompiled RPMS this is the wish thinking to have in the future. DriverContainers could be easily built just by using RPMs. 
+
+Some drivers will *never* be upstreamed and can be in any of the states described above.  The proposed solutions needs to handle drivers in any state and build by "any" tool. 
+
+The compilation of kernel modules was always anticipated to be the fallback solution when dealing with kernel modules. Some kernel modules will always be out-of-tree and are not going to be included in the near future. Some kernel modules are out-of-tree but we are working with the vendors on upstreaming them  to the mainline kernel. 
+This work is being done by a virtual-team consisting of an  OpenShift-accelerators  and RHEL-accelerator team responsible for enabling hardware and software on RHEL and OpenShift. 
+
+
+### Goals
+
+
+* A unified way to deploy out-of-tree drivers on OpenShift 4.x on all supported Red Hat Operating Systems
+* The solution should avoid rebuilds on every node and allow for distribution of drivers on a cluster using the cluster registry
+* A solution for day-1 and day-2 kernel modules
+* Support upgrades of OpenShift Support of multiple kernel module providers
+* Hierarchical initialization of kernel modules (modeling dependencies)
+*  Handle dependencies between kernel modules (depmod) in tree and out of tree
+* Should support disconnected and proxy environments
+* Support building a DriverContainer on any host or kernel version (-k flags) e.g CI
+* Support heterogeneous cluster: 
+  * OpenShift with RHEL7, 8 and RHCOS 
+  * OKD is only supporting FCOS nodes for now. 
+
+
+### Non-Goals
+
+* The solution is not a replacement for the traditional way of delivering kernel modules, customer/partners should be aware that we prefer they  deliver the drivers upstream
```

Comment:
This statement is vague and needs to be clearly scoped. The "traditional way of delivering kernel modules" is a positional statement and lacks specficitity of the target. For example, the traditional way for Linux in general, RHCOS, RHEL, out-of-tree support, etc.  Perhaps:
```
* Delivery of kernel modules outside the scope of OpenShift
```